File: csharp/Practice01-Begin/src/MyClient/MyConnection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace MyClient
{
    public class MyConnection : IDisposable
    {
        public readonly static int ReconnectInterval = 3000;

        public MyConnection(string[] uris)
        {
            throw new NotImplementedException();
        }

        public void Open()
        {
            throw new NotImplementedException();
        }

        public void Dispose()
        {
            throw new NotImplementedException();
        }

        public int Subscribe(IMySubscriber subscriber)
        {
            throw new NotImplementedException();
        }

        public bool Unsubscribe(int subscriptionId)
        {
            throw new NotImplementedException();
        }

        public event EventHandler Connected;

        public event EventHandler ConnectFailed;

        public event EventHandler Disconnected;
    }
}

File: java/Practice01-Begin/src/myClient/MyConnection.java
package myClient;

import java.io.Closeable;

public class MyConnection implements Closeable {
    
	public final static int RECONNECT_INTERVAL = 3000;
	
    public MyConnection(String[] uris) {
        throw new RuntimeException("Not implemented");
    }
    
    public void open() {
        throw new RuntimeException("Not implemented");
    }
    
    public void close() {
        throw new RuntimeException("Not implemented");
    }

    public int subscribe(MySubscriber subscriber) {
        throw new RuntimeException("Not implemented");
    }
    
    public boolean unsubscribe(int subscriptionId) {
    	throw new RuntimeException("Not implemented");
    }
    
    public synchronized void addConnectionListener(MyConnectionEventListener listener) {
        throw new RuntimeException("Not implemented");
    }
    
    public synchronized void removeConnectionListener(MyConnectionEventListener listener) {
        throw new RuntimeException("Not implemented");
    }
}
